IPython/lib/pretty.py

```diff
@@ -623,45 +708,38 @@ def _super_pprint(obj, p, cycle):
     p.end_group(8, '>')
 
 
-def _re_pattern_pprint(obj, p, cycle):
-    """The pprint function for regular expression patterns."""
-    p.text('re.compile(')
-    pattern = repr(obj.pattern)
-    if pattern[:1] in 'uU':
-        pattern = pattern[1:]
-        prefix = 'ur'
-    else:
-        prefix = 'r'
-    pattern = prefix + pattern.replace('\\\\', '\\')
-    p.text(pattern)
-    if obj.flags:
-        p.text(',')
-        p.breakable()
+
+class _ReFlags:
+    def __init__(self, value):
+        self.value = value
+
+    def _repr_pretty_(self, p, cycle):
         done_one = False
         for flag in ('TEMPLATE', 'IGNORECASE', 'LOCALE', 'MULTILINE', 'DOTALL',
             'UNICODE', 'VERBOSE', 'DEBUG'):
-            if obj.flags & getattr(re, flag):
+            if self.value & getattr(re, flag):
                 if done_one:
                     p.text('|')
                 p.text('re.' + flag)
                 done_one = True
-    p.text(')')
+
+
+def _re_pattern_pprint(obj, p, cycle):
+    """The pprint function for regular expression patterns."""
+    re_compile = CallExpression.factory('re.compile')
+    if obj.flags:
+        p.pretty(re_compile(RawStringLiteral(obj.pattern), _ReFlags(obj.flags)))
+    else:
+        p.pretty(re_compile(RawStringLiteral(obj.pattern)))
```

Comment:
Or just wondering if the factory is _that_ needed to avoid a small duplication of code ?

```suggestion
def _re_pattern_pprint(obj, p, cycle):
    """The pprint function for regular expression patterns."""
    if obj.flags:
        p.pretty(CallExpression('re.compile', RawStringLiteral(obj.pattern), _ReFlags(obj.flags)))
    else:
        p.pretty(CallExpression('re.compile', RawStringLiteral(obj.pattern)))
```

?

Or am I missunderstanding ?

Reply:
That's a actually what I started with, but thought I'd play with the syntactic sugar